### PR TITLE
BE/FE: Get User Route + Stay Logged-In

### PR DESCRIPTION
**Pull request by Hakeemmidan**
In this pull request, I:
- Created an API route for getting the current logged in user
  - `GET /users/current`
- Fetched logged in user on mount of `App` (frontend) and automatically authenticated them if they had a valid `jwt`

#### Frontend:
1. Log in or register
2. Close tab or browser
3. Go back to app
4. Check that you are still logged in

You should stay logged in

#### Backend:
- `yarn test`

All tests should pass

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -16,6 +15,7 @@ function App() {
 
   React.useEffect(() => {
     // Verify if the user is validated and if so, setUser
+    if (!user._id) user.getCurrent();
   });
```
Comment (tgnemecek) [inline client/src/App.js:19]: This function will run on every render, ideally we only want to run it on component mount. To do that with useEffect you provide an empty array as the second argument. If you want I can explain how this works.
```suggestion
  }, []);
```

File: server/routes/auth/auth.route.js [hunk 2]
```diff
@@ -34,10 +34,17 @@ router.post('/register', (req, res) => {
         .then((user) => {
           const payload = { id: user.id };
```
Comment (tgnemecek) [inline server/routes/auth/auth.route.js]:
```suggestion
          const payload = { id: user._id };
```

File: server/routes/auth/auth.route.js [hunk 3]
```diff
@@ -34,10 +34,17 @@ router.post('/register', (req, res) => {
...
           const token = user.signJWT(payload);
+          const options = {
+            httpOnly: true,
```
Comment (tgnemecek) [inline server/routes/auth/auth.route.js:38]: Don't you need to set the expiration here too? Here we previously had:
`expires: new Date(
              Date.now() + tokenExpiration * 24 * 60 * 60 * 1000
            ),`

File: server/routes/auth/auth.route.js [hunk 4]
```diff
@@ -68,10 +75,17 @@ router.post('/login', (req, res) => {
     if (isMatch) {
       const payload = { id: user.id };
       const token = user.signJWT(payload);
+      const options = {
+        httpOnly: true,
```
Comment (tgnemecek) [inline server/routes/auth/auth.route.js:79]: Same question about the expiration here.

**Review by tgnemecek — APPROVED** [review]
I had a few comments on minor things, but overall this looks great! Keep it up!

File: client/src/App.js (hunk 1 above)
Comment (Hakeemmidan) [reply client/src/App.js:19]: resolved in #54

File: server/routes/auth/auth.route.js (hunk 3 above)
Comment (Hakeemmidan) [reply server/routes/auth/auth.route.js:38]: I believe that once the `jwt` token expires (expiration listed in `models/user.js`), `passport` should dis-authenticate the user and asks them to login/register. So having it just there should be fine.

**Pull request merged**